simulator/mips/mips_instr.h

```diff
@@ -282,16 +280,27 @@ class FuncInstr
             if ( _is_jump_taken)
                 new_PC += sign_extend( v_imm) << 2;
         }
-        
+
         void execute_clo() { v_dst = count_zeros( ~v_src1); }
         void execute_clz() { v_dst = count_zeros(  v_src1); }
-        
+
         void execute_j()      { _is_jump_taken = true; new_PC = (PC & 0xf0000000) | (v_imm << 2); }
         void execute_jr()     { _is_jump_taken = true; new_PC = align_up<2>(v_src1); }
 
         void execute_jal()    { _is_jump_taken = true; v_dst = new_PC; new_PC = (PC & 0xF0000000) | (v_imm << 2); };
         void execute_jalr()   { _is_jump_taken = true; v_dst = new_PC; new_PC = align_up<2>(v_src1); };
 
+        template<Predicate p>
+        void execute_jlr()
```

Comment:
Could you please rename it to "execute_branch_and_link"? MIPS uses "jump" name for unconditional jumps (both direct and indirect).